huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go

```diff
@@ -418,20 +441,66 @@ func resourceDmsKafkaInstanceCreate(ctx context.Context, d *schema.ResourceData,
 	return resourceDmsKafkaInstanceRead(ctx, d, meta)
 }
 
-func resourceDmsKafkaInstanceRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func flattenConnectPorts(strInfos string) ([]map[string]interface{}, error) {
+	defer func() {
+		if r := recover(); r != nil {
+			log.Printf("[ERROR] Recover panic when flattening cross-VPC infos structure: %#v", r)
+		}
+	}()
+
+	crossVpcInfos := make(map[string]interface{})
+	err := json.Unmarshal([]byte(strInfos), &crossVpcInfos)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]map[string]interface{}, 0, len(crossVpcInfos))
+	for _, val := range crossVpcInfos {
+		crossVpcInfo := val.(map[string]interface{})
+		result = append(result, map[string]interface{}{
+			// The address of the crossVpcInfos is the same as the addresses key, so drop the key.
```

Comment:
suggest to keep what it is as API/Console side. add lisenter_ip and advertised_ip

Reply:
done.